utils/doclint/xmlDocumentation.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+const Documentation = require('./documentation');
+const { visitAll } = require('../markdown');
+/**
+ * @param {Documentation.MarkdownNode[]} nodes
+ * @param {number} maxColumns
+ */
+function renderXmlDoc(nodes, maxColumns) {
+    const summary = [];
+    const examples = [];
+    /** @type {Documentation.MarkdownNode} */
+    let lastNode;
+
+    summary.push('<summary>');
+
+    visitAll(nodes, (node, depth) => {
+        lastNode = innerRenderXmlNode(node, lastNode, summary, examples, maxColumns);
+    });
+
+    if (summary.length == 1) { // just the <summary> node 
+        return [];
+    }
+
+    // we might have a stray list, if the <li> is the last element
+    if (lastNode && lastNode.type === 'li') {
```

Comment:
It looks like md's list items differ from the xmldoc ones in a way that it wants a wrapping container. I would render lists as md, i.e. via prepending `- `. That'll erase this `lastNode` handling.

Reply:
So, VS isn't _smart_ enough to actually render that in a meaningful way. So I'll always need to wrap it in a `<list>` if I want it to display at least somewhat nicely to the end user.

This:

```csharp
    /// <summary>
    /// Remember, that sometimes, you need to do the following:
    /// - Test 1
    /// - Test 2
    /// - Test 3
    /// - Test 4
    /// 
    /// But there are times, when you want to do
    /// <list type="bullet">
    /// <item><description>Something smart</description></item>
    /// <item><description>Something smarter</description></item>
    /// <item><description>The really smart thing</description></item>
    /// <item><description>Just a list</description></item>
    /// </list>
    /// Tis a thing <seealso 
    /// cref="Browser"/> test.
    /// <a href="https://www.google.com">This is a link.</a>
    /// </summary>
```
turns into this:

![image](https://user-images.githubusercontent.com/1932563/106108585-bceb2a00-6148-11eb-87c6-2b41533431d1.png)